vyper/utils.py

```diff
@@ -138,8 +138,6 @@ def is_varname_valid(varname):
         return False
     if varname.lower() in reserved_words:
         return False
-    if varname[0] == '~':
```

Comment:
Why are you removing this check?

Reply:
Nice

Reply:
No need to test illegal syntax! Python core already does it for us!

Reply:
Wait, that `ast` error isn't quite what I expected. Can you potentially use it outside of instantiation (where type hinting doesn't come into play), e.g. in a function name or something?